include/yaml-cpp/node/impl.h

```diff
@@ -209,6 +240,19 @@ inline Node& Node::operator=(const Node& rhs) {
   return *this;
 }
 
+inline Node& Node::operator=(Node&& rhs) YAML_CPP_NOEXCEPT {
+  if (is(rhs))
+    return *this;
+  Node tmp(std::move(rhs));
+  try {
+    AssignNode(tmp);
+  }
+  catch(...) {
```

Comment:
I think this is a little unclear. The problem seems to be that AssignNode throws InvalidNodeException? Maybe this needs to be refactored something like

CheckValid()

which does the if (isValid)/throw check

EnsureNodeExists

which does the rest, and then AssignNode can just be inlined into this operator= and avoid throwing.

Reply:
Yes, in order to be able to make it `noexcept` I was trying to come up with something, but couldn't figure out how to do it without calling some function that could throw. Now that I'm looking at it, I see that I missed the throw in `is()`. I'll try to make something out of your suggestions.